addon/plugins/validation/utils/transaction-complies-with-shapes.ts

```diff
@@ -0,0 +1,28 @@
+import { EditorState, Transaction } from '@lblod/ember-rdfa-editor';
+import { validateTransaction } from '@lblod/ember-rdfa-editor-lblod-plugins/plugins/validation';
+
+/**
+ * Validate a given transaction against a given set of shapes. Returns true when the
+ * state after applying the transaction is valid according to all given shapes,
+ * false otherwise.
+ * @param state the state the transaction was created from
+ * @param tr
+ * @param validateShapes
+ */
+export function transactionCompliesWithShapes(
+  state: EditorState,
+  tr: Transaction,
+  validateShapes?: Set<string>
+) {
+  if (validateShapes?.size) {
+    const report = validateTransaction(state, tr);
+    if (
+      report.results?.some((result) =>
+        validateShapes.has(result.sourceShape.name)
+      )
+    ) {
+      return false;
+    }
+  }
+  return true;
+}
```

Comment:
yeah that was exactly my first implementation :)
but then you have the situation where theres two validation errors, and the transaction solves only one
(e.g. both title and description are missing, and you're trying to add a title)
-> suddenly the "add-title" command can't run because "after running" there are still validation errors

And yeah I _could_ check if the _amount_ of validations is reduced, but that's also not completely sound, since a transaction may inadvertently fix a different validation than it's supposed to, or only fix one of the multiple validations it's intended to fix